src/components/molecules/TableComponent/TableComponent.tsx

```diff
@@ -20,51 +33,146 @@ const TableComponent: React.FunctionComponent<TableComponentPropsType> = ({
   const locked = tableLocked(table.reference);
   const numberOfSeatsLeft = table.capacity && table.capacity - users.length;
   const full = numberOfSeatsLeft === 0;
+  const { userId } = useUser();
+
+  const { isAdminUser, isLoading: isCheckingRole } = useIsAdminUser(userId);
+
+  const isRemoveButtonShown = !isCheckingRole && isAdminUser;
+
+  const {
+    isShown: isModalShown,
+    toggle: toggleModal,
+    hide: hideModal,
+  } = useShowHide(false);
+
+  const [{ loading: isDeletingTable }, removeTable] = useAsyncFn(async () => {
+    if (!venue.id) return;
+
+    await deleteTable({
+      venueId: venue.id,
+      tableName: table.reference,
+    });
+  }, [table.reference, venue.id]);
+
+  const [{ loading: isUpdatingTables }, updateTables] = useAsyncFn(async () => {
+    if (!venue.id) return;
+
+    await updateVenueTable({
+      venueId: venue.id,
+      defaultTables: CONVERSATION_TABLES,
+    });
+  }, [venue.id]);
+
+  const handleTableDelete = async () => {
+    // there are venues that don't have tables inserted in the database and use default ones instead
+    // if that's the case, then we won't be able to delete any of the tables
+    // thus we have to create tables before modifying them
```

Comment:
I've moved table update logic into `removeTable` cloud function and got rid of `JazzBarTableComponent` entirely, so that now there's only one table component bd10ed1